src/assets/scripts/client/game/ScoreController.js

```diff
@@ -0,0 +1,253 @@
+     * @private
+     */
+    _onAirspaceExitWithClearance(aircraftModel) {
+        aircraftModel.radioCall('switching to center, good day', AIRPORT_CONTROL_POSITION_NAME.DEPARTURE);
+        GameController.events_recordNew(GAME_EVENTS.DEPARTURE);
+    }
+
+    /**
+     * @for ScoreController
+     * @method _onAirspaceExitWithoutClearance
+     * @param aircraftModel {AircraftModel}
+     * @private
+     */
+    _onAirspaceExitWithoutClearance(aircraftModel) {
+        aircraftModel.radioCall('leaving radar coverage without proper clearance', AIRPORT_CONTROL_POSITION_NAME.DEPARTURE, true);
+        GameController.events_recordNew(GAME_EVENTS.NOT_CLEARED_ON_ROUTE);
+    }
+
+    /**
+     * @for ScoreController
+     * @method _scoreWind
+     * @param aircraftModel {AircraftModel}
+     * @param action {string}
+     */
+    _scoreWind(aircraftModel, action) {
+        const isWarning = true;
+        const wind = aircraftModel.getWind();
+
+        // TODO: these two if blocks could be done in a single switch statement
+        if (wind.cross >= 20) {
+            GameController.events_recordNew(GAME_EVENTS.EXTREME_CROSSWIND_OPERATION);
+            UiController.ui_log(`${aircraftModel.callsign} ${action} with major crosswind`, isWarning);
+        } else if (wind.cross >= 10) {
+            GameController.events_recordNew(GAME_EVENTS.HIGH_CROSSWIND_OPERATION);
+            UiController.ui_log(`${aircraftModel.callsign} ${action} with crosswind`, isWarning);
+        }
+
+        if (wind.head <= -10) {
+            GameController.events_recordNew(GAME_EVENTS.EXTREME_TAILWIND_OPERATION);
+            UiController.ui_log(`${aircraftModel.callsign} ${action} with major tailwind`, isWarning);
+        } else if (wind.head <= -5) {
+            GameController.events_recordNew(GAME_EVENTS.HIGH_TAILWIND_OPERATION);
+            UiController.ui_log(`${aircraftModel.callsign} ${action} with tailwind`, isWarning);
+        }
+    }
+
+    /**
+     * @for ScoreController
+     * @method _scoreRunwaySeparation
+     * @param aircraftModel {AircraftModel}
+     * @param runwayModel {RunwayModel}
+     * @param action {string}
+     */
+    _scoreRunwaySeparation(aircraftModel, runwayModel, action) {
+        const previousAircraft = this._aircraftController.findAircraftByCallsign(runwayModel.lastDepartedAircraftId);
+
+        if (!previousAircraft) {
+            return;
+        }
+
+        const actualDistance = nm_ft(aircraftModel.distanceToAircraft(previousAircraft));
+        const requiredDistance = aircraftModel.model.calculateSameRunwaySeparationDistanceInFeet(previousAircraft.model);
+
+        if (actualDistance < requiredDistance || previousAircraft.isOnGround()) {
+            const isWarning = true;
+
+            GameController.events_recordNew(GAME_EVENTS.NO_TAKEOFF_SEPARATION);
+            UiController.ui_log(`${aircraftModel.callsign} ${action} while another aircraft was using the same runway`, isWarning);
```

Comment:
Slight tweak, if you could: `without adequate separation from another aircraft using the same runway`

Reply:
done